plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+            self.validate_traffic_type(traffic_type.upper())
+
+        vlan_id = anycast.get("vlan_id")
+        if vlan_id and vlan_id not in range(2, 4094) or vlan_id in [1002, 1003, 1004, 1005, 2046]:
+            self.msg = (
+                "Invalid vlan_id '{0}' given in the playbook. Allowed VLAN range is (2,4094) except for "
+                "reserved VLANs 1002-1005, and 2046."
+            ).format(vlan_id)
+            self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
+
+        self.log("Given parameter '{0}' for the configuration of anycast gateway validated successfully.".format(str(anycast)), "INFO")
+
+        return self
+
+    def get_anycast_gateway_mapping(self, vn_name):
+        """
+        Retrieves a mapping of Anycast Gateway configuration parameters from their common names
+        to their respective API field names.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            vn_name (str): The name of the layer3 Virtual Network to check whether it's INFRA_VN or not
+                and seperate the payload for the API for respective VN.
+        Returns:
+            dict: A dictionary where the keys are common parameter names used in configuration
+                and the values are the corresponding field names used in the Anycast Gateway API.
+        Description:
+            This function creates and returns a mapping of various configuration parameters
+            associated with Anycast Gateways. The mapping translates common, user-friendly names
+            into the specific field names expected by the API.
+        """
+
+        gateway_mapping = {
+            "tcp_mss_adjustment": "tcpMssAdjustment",
+            "vlan_name": "vlanName",
+            "vlan_id": "vlanId",
+            "traffic_type": "trafficType",
+            "pool_type": "poolType",
+            "security_group_name": "securityGroupName",
+            "is_critical_pool": "isCriticalPool",
+            "layer2_flooding_enabled": "isLayer2FloodingEnabled",
+            "fabric_enabled_wireless": "isWirelessPool",
+            "ip_directed_broadcast": "isIpDirectedBroadcast",
+            "intra_subnet_routing_enabled": "isIntraSubnetRoutingEnabled",
+            "multiple_ip_to_mac_addresses": "isMultipleIpToMacAddresses",
+            "supplicant_based_extended_node_onboarding": "isSupplicantBasedExtendedNodeOnboarding",
+            "group_policy_enforcement_enabled": "isGroupBasedPolicyEnforcementEnabled"
+        }
+
+        if vn_name == "INFRA_VN":
+            params_to_remove = ["is_critical_pool", "layer2_flooding_enabled", "fabric_enabled_wireless", "security_group_name",
+                                "ip_directed_broadcast", "intra_subnet_routing_enabled", "multiple_ip_to_mac_addresses"]
+            for item in params_to_remove:
+                gateway_mapping.pop(item, None)
+
+        return gateway_mapping
```

Comment:
self.log("Final gateway mapping for '{0}': {1}".format(vn_name, gateway_mapping), "INFO")
    return gateway_mapping